nifi-nar-bundles/nifi-splunk-bundle/nifi-splunk-processors/src/main/java/org/apache/nifi/processors/splunk/QuerySplunkIndexingStatus.java

```diff
@@ -234,4 +244,18 @@ private String generateContent(final Map<Long, FlowFile> undetermined) throws IO
             return Optional.empty();
         }
     }
+
+    private static boolean isAlreadyChecked(FlowFile flowFile) {
+        return extractBoolean(flowFile.getAttribute(ACK_CHECKED_ATTRIBUTE));
+    }
+
+    private static boolean extractBoolean(final String value) {
+        return Boolean.parseBoolean(value);
+    }
+
+    private void setAckCheckedToTrue(final ProcessSession session, final FlowFile flowFile) {
+        final Map<String, String> attributes = new HashMap<>(flowFile.getAttributes());
+        attributes.put(ACK_CHECKED_ATTRIBUTE, String.valueOf(Boolean.TRUE));
+        session.putAllAttributes(flowFile, attributes);
```

Comment:
```suggestion
        return session.putAttribute(flowFile, ACK_CHECKED_ATTRIBUTE, Boolean.toString(true));
```
`putAttribute` (and `putAllAttribute`) are _not_ void methods.
Shouldn't we work with the returned `flowFile` in the later segments of the code?

Reply:
You are correct, FFs are immutable by design.